src/components/WalletModal/index.tsx

```diff
@@ -135,48 +137,48 @@ const ModalContent = (props: any) => {
 
     const { ethereum } = window
     const toggleNetworkModal = useNetworkModalToggle()
-
+    const isMultiple = useSelectedProvider()
     const handleEthereumNetworkSwitch = useCallback(() => {
         const networkType = process.env.REACT_APP_NETWORK || 'staging'
         if (networkType === 'staging') {
             toggleNetworkModal()
         } else if (networkType === 'production') {
-            ;(ethereum as any)?.request({
+              ;(isMultiple ? ethereum?.selectedProvider as any : ethereum as any)?.request({
                 method: 'wallet_switchEthereumChain',
                 params: [{ chainId: `0x${ChainId.MAINNET.toString(16)}` }]
-            })
+              })
             toggleWalletModal()
         }
     }, [ethereum, toggleNetworkModal, toggleWalletModal])
 
     const handleFuseNetworkSwitch = useCallback(() => {
-        ;(ethereum as any)?.request({
-            method: 'wallet_addEthereumChain',
-            params: [
-                {
-                    chainId: '0x7a',
-                    chainName: 'Fuse',
-                    nativeCurrency: {
-                        name: 'FUSE Token',
-                        symbol: 'FUSE',
-                        decimals: 18
-                    },
-                    rpcUrls: ['https://rpc.fuse.io'],
-                    blockExplorerUrls: ['https://explorer.fuse.io']
+      ;(isMultiple ? ethereum?.selectedProvider as any : ethereum as any).request({
```

Comment:
what if it is not multiple where's the call to add chain?

Reply:
see above

Reply:
same answer as above for the multiple

second, add chain is not necessary I guess? as eth is a network that always exists 
(it can't even be removed from meta / coinbase)